api/graphql/mutations/network.js

```diff
@@ -2,12 +2,50 @@ import validateNetworkData from '../../models/network/validateNetworkData'
 import underlyingUpdateNetwork from '../../models/network/updateNetwork'
 import convertGraphqlData from './convertGraphqlData'
 
-export function updateNetwork (userId, { id, data }) {
+// TODO: more integrated `isAdmin` handling for mutations?
+export async function networkMutationPermissionCheck ({ userId, isAdmin = false }, networkId) {
+  if (isAdmin) return
+  if (!await NetworkMembership.hasModeratorRole(userId, networkId)) {
+    throw new Error("You don't have permission to modify that network.")
+  }
+}
+
+export async function addCommunityToNetwork (user, { communityId, networkId }) {
+  await networkMutationPermissionCheck(user, networkId)
```

Comment:
I find it misleading to name this first argument `user`, since it's not a user object.

Reply:
Yeah, I struggled with that too. I tried 'credentials' but it seemed weird. Here's a try with `authZ` (vs authN), see if that makes sense to you.